presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -239,6 +249,34 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
         }
     }
 
+    private void checkPermissions(Session session)
+    {
+        // When selectAuthorizedIdentity API is not enabled, only check delegation permission
+        if (!securityConfig.isAuthorizedIdentitySelectionEnabled()) {
+            accessControl.checkCanSetUser(
+                    session.getIdentity(),
+                    new AccessControlContext(
+                            session.getQueryId(),
+                            session.getClientInfo(),
+                            session.getSource()),
+                    session.getIdentity().getPrincipal(),
+                    session.getIdentity().getUser());
+        }
+
+        // When selectAuthorizedIdentity API is enabled,
+        // 1. Check the delegation permission, which is inside the API call
+        // 2. Select the authorized user
+        // 3. TODO: UPDATE the identity of the session according to the authorized identity from the API call
+        accessControl.selectAuthorizedIdentity(
```

Comment:
This returns an identity, but the return value is never used. If this method is supposed to validate and throw, change the method checkAuthorizedIdentity.

Reply:
We need to change the name of this function. If you use IDEA, they should be easy to change I assume.